### ISSUE TITLE
Display all request headers in Timeline tab

### ISSUE THREAD

**Issue by jackc** [usebruno/bruno#645]
[JIRA](https://usebruno.atlassian.net/browse/BRU-3731)

I have a Rails API. I've been using curl to for ad hoc testing, but just tried using Bruno. Unfortunately, the experience was very confusing.

A Rails API that only handles json will typically work with or without an "Accept: application/json" header. So when using curl I do not include it. I did the same thing with Bruno and received an HTTP 406 Not Acceptable. I was unable to find any difference in the Bruno UI. However, I then pointed Bruno at a netcat listener so I could see **exactly** what was going over the wire. It is sending a default header of `accept: application/json, text/plain, */*`.

Further digging revealed that Rails treats any accept header including `*/*` as coming from a browser and essentially overrides the accept header to only respond with html. (https://github.com/rails/rails/issues/40908 and https://github.com/rails/rails/blob/ae6fa7a048d0a0f79b0f8dd8566c7d499fd6f616/actionpack/lib/action_dispatch/http/mime_negotiation.rb#L171)

The underlying issue was with Rails. However, it was more difficult to diagnose because Bruno does not appear to expose all the actual request headers. The "Timeline" tab appears to show request headers, but it actually only shows the explicitly set headers. This was misleading. I wasn't expecting an accept header to be set at all, but it was happening invisibly.

Perhaps all request headers could be displayed in the Timeline view.

**Comment by pete-mantell-giffgaff** [comment]
I'd like this too - or alternatively extra tabs on the RHS which show _actual_ request body & headers sent (i.e. after all scripts, vars, etc are evaluated) to assist with diagnosis

**Comment by palhal** [comment]
Yes, this is a must-have in my opinion. I'm currently debugging why the API behaves differently when using Bruno vs. application code. Then it's crucial to know *all* headers that are sent.

**Comment by bannmann** [comment]
Same here. I was curious what the user agent value sent by Bruno was. Another use case I had a few months ago (in Insomnia) is debugging a problem with cookies.

Bruno should follow Insomnia's lead here and simply display **all** headers with their **final values** in the timeline tab.

**Comment by tjalling-ran** [comment]
Same here. It applies to request parameters as well.

Not having this makes debugging harder than it needs to be.

**Comment by andrei-korshikov** [comment]
> I was curious what the user agent value sent by Bruno was.

Same here. And finally I found out way more things than I wanted to know…

All related issues (for the record): #137, #1225, #1669, #2037.

**Comment by helloanoop** [comment]
This is being worked upon and should be available in a couple of weeks. Here is a snapshot of the work so far on enhancing the timeline view.

<img width="700" alt="Image" src="https://github.com/user-attachments/assets/039e280c-57ed-4eef-9ead-584f9d2da551" />

<img width="700" alt="Image" src="https://github.com/user-attachments/assets/3eab1e6e-a202-44d9-adfb-eaf84c41f98e" />

**Comment by anusree-bruno** [comment]
The new timeline view is available in the later versions. Closing this as resolved.

**Comment by nwin** [comment]
@anusree-bruno when is it available? I am still seeing this issue with 2.9.1.

**Comment by anusree-bruno** [comment]
All headers are visible in the Timeline tabs. If you’re still not seeing some headers, could you share a screenshot of your view so we can check for any discrepancy?

<img width="1541" height="691" alt="Image" src="https://github.com/user-attachments/assets/0bb7bb43-6877-4573-875e-a3b95e2e4faf" />

**Comment by nwin** [comment]
There are some headers missing

<img width="1905" height="707" alt="Image" src="https://github.com/user-attachments/assets/b3fbab80-0c37-40d0-8e78-796f317003f0" />

**Comment by phillycheeze** [comment]
Anyone discover a way to actually see the full request headers being sent? Like @nwin 's screenshot shows, some of the http spec default headers aren't shown in the timeline view or the dev tools view.

**Comment by andrei-korshikov** [comment]
@phillycheeze as been said many times before—use curl or [Bruno Lazer](https://github.com/Its-treason/bruno)

**Comment by phillycheeze** [comment]
@andrei-korshikov Been said many times where? And saying "use another tool" isn't helpful but thanks for the reply.

**Comment by andrei-korshikov** [comment]
> Been said many times where?

Here, in [Bruno issues](https://github.com/usebruno/bruno/issues).

> And saying "use another tool" isn't helpful

O_O

**Comment by anusree-bruno** [comment]
We’re fixing this issue along with an updated UI. Here’s a little sneak peek at what we’re building. Would love to hear your feedback.

<img width="4064" height="2242" alt="Image" src="https://github.com/user-attachments/assets/6376a9cf-c914-4f50-90c1-8c731e2510a5" />

**Comment by anusree-bruno** [comment]
This issue has been fixed in v4.2.0

We have added UI to view and manage headers added by bruno at runtime as well as updated the timeline to show all the headers sent.
